recipes-bsp/uefi/tegra-uefi-capsules_35.5.0.bb

```diff
@@ -11,45 +11,84 @@ TEGRA_UEFI_CAPSULE_SIGNING_EXTRA_DEPS ??= ""
 
 COMPATIBLE_MACHINE = "(tegra)"
 
-DEPENDS += "tegra-bup-payload"
+TEGRA_SIGNING_EXTRA_DEPS ??= ""
 
 GUID:tegra194 ?= "be3f5d68-7654-4ed2-838c-2a2faf901a78"
 GUID:tegra234 ?= "bf0d4599-20d4-414e-b2c5-3595b1cda402"
 
 do_compile() {
+    # Generate BUP images
+    PATH="${STAGING_BINDIR_NATIVE}/${FLASHTOOLS_DIR}:${PATH}"
+    export tosimgfilename=${TOSIMGFILENAME}
+    rm -rf ${WORKDIR}/bup-payload
+    mkdir ${WORKDIR}/bup-payload
+    oldwd="$PWD"
+    cd ${WORKDIR}/bup-payload
+    # BUP generator really wants to use 'boot.img' for the LNX
+    # partition contents
+    tegraflash_populate_package ${IMAGE_TEGRAFLASH_KERNEL} boot.img ${@tegra_bootcontrol_overlay_list(d, bup=True)}
+    mv generate_bup_payload.sh doflash.sh
+    tegraflash_create_flash_config flash.xml.in boot.img ${STAGING_DATADIR}/tegraflash/bupgen-internal-flash.xml
+    . ./flashvars
+    tegraflash_custom_sign_bup
+    for bup in ${WORKDIR}/bup-payload/${BUP_PAYLOAD_DIR}/*; do
+	    [ -e $bup ] || continue
+	    BUP_generator.py --contents --check $bup
+    done
+    mv ${WORKDIR}/bup-payload/${BUP_PAYLOAD_DIR}/* .
+    cd "$oldwd"
+
+    # Deploy BUP payloads
+    for f in ${WORKDIR}/bup-payload/*_only_payload; do
+        [ -e $f ] || continue
+        sfx=$(basename $f _payload)
+        # We need to have bup-payload images to be deployed to ${DEPLOY_DIR_IMAGE} before
```

Comment:
what explained in this comment - is how it was done in the previous implementation, where CONVERSION_CMD was used. TBH I'd prefer to split that recipe on two, and add compile time dependency, so capsule-generation recipe depends bup-recipe

This all applies if we still want to keep `sign_uefi_capsules` function implementation at it is (as it expects BUP images to be deployed to DEPLOY_DIR_IMAGE in advance)

Reply:
@madisongh @Lexmark-chad please advise what action needs to be taken

Reply:
@madisongh sounds good to me. 

Should we still deploy BUP images to `${DEPLOYDIR}` in `do_deploy()` task, or they are not supposed to used anymore?

I was thinking about generating both BUP images and capsules in `${B}` dir, but then deploying all of them in `do_deploy()`

Reply:
BUPs aren't supposed to be used any more, so let's not deploy them, just the capsules.

Reply:
Totally agree with @madisongh. Let's use only capsules 
Thanks @igoropaniuk

Reply:
done!